packages/@react-aria/combobox/src/useComboBox.ts

```diff
@@ -15,7 +15,7 @@ import {AriaButtonProps} from '@react-types/button';
 import {ariaHideOutside} from '@react-aria/overlays';
 import {AriaListBoxOptions, getItemId, listData} from '@react-aria/listbox';
 import {chain, isAppleDevice, mergeProps, useLabels} from '@react-aria/utils';
-import {ComboBoxProps} from '@react-types/combobox';
+import {AriaComboBoxProps as ComboBoxProps} from '@react-types/combobox';
```

Comment:
Im using an alias here instead of renaming everything below.

Reply:
IMO for consistency you should follow the pattern setup by useSelect: https://github.com/adobe/react-spectrum/blob/a98461602538bcf63009a17e31900726819a4af6/packages/%40react-aria/select/src/useSelect.ts#L26-L32

Essentially props that are specific to the aria hook and shouldn't be exposed in the RSP component props should go inside a `Aria[COMPONENTNAME]Options` interface that extends off of the `Aria[COMPONENTNAME]Props`

Perhaps something like
```
import {AriaComboBoxProps} from '@react-types/combobox';
...

interface AriaComboBoxOptions<T> extends AriaComboBoxProps<T> {
  /** The ref for the input element. */
  inputRef: RefObject<HTMLInputElement>,
  /** The ref for the list box popover. */
  popoverRef: RefObject<HTMLDivElement>,
  /** The ref for the list box. */
  listBoxRef: RefObject<HTMLElement>,
  /** The ref for the optional list box popup trigger button.  */
  buttonRef?: RefObject<HTMLElement>,
  /** An optional keyboard delegate implementation, to override the default. */
  keyboardDelegate?: KeyboardDelegate
}
....

export function useComboBox<T>(props: AriaComboBoxOptions<T>, state: ComboBoxState<T>): ComboBoxAria<T> {
...

```